bftengine/src/bftengine/DbCheckpointManager.cpp

```diff
@@ -199,15 +184,17 @@ void DbCheckpointManager::initializeDbCheckpointManager(std::shared_ptr<concord:
     init();
   } else {
     // db checkpoint is disabled. Cleanup metadata and checkpoints created if any
-    auto ret = std::async(std::launch::async, [this]() { cleanUp(); });
+    cleanUpFuture_ = std::async(std::launch::async, [this]() { cleanUp(); });
   }
 }
 
-void DbCheckpointManager::createDbCheckpointAsync(const SeqNum& seqNum, const std::optional<Timestamp>& timestamp) {
-  auto ret = std::async(std::launch::async, [this, seqNum, timestamp]() -> void {
-    const auto& lastBlockid = getLastBlockIdCb_();
+uint64_t DbCheckpointManager::createDbCheckpointAsync(const SeqNum& seqNum, const std::optional<Timestamp>& timestamp) {
+  const auto lastBlockid = getLastBlockIdCb_();
+  auto ret = std::async(std::launch::async, [this, seqNum, timestamp, lastBlockid]() -> void {
     createDbCheckpoint(lastBlockid, lastBlockid, seqNum, timestamp);
   });
+  dbCreateCheckPtFuture_.emplace(std::make_pair(lastBlockid, std::move(ret)));
```

Comment:
Shouldn't we hold the `lock` here, for updating `dbCreateCheckPtFuture_`? We call `removeOldestDbCheckpoint()` in `createDbCheckpoint()` and that is in a separate thread.